plugins/module_utils/gitlab.py

```diff
@@ -110,3 +110,14 @@ def gitlab_authentication(module):
             GitLab remove Session API now that private tokens are removed from user API endpoints since version 10.2." % to_native(e))
 
     return gitlab_instance
+
+
+def filter_returned_variables(gitlab_variables):
+    # pop properties we don't know
+    existing_variables = [x.attributes for x in gitlab_variables]
+    KNOWN = ['key', 'value', 'masked', 'protected', 'variable_type', 'environment_scope']
+    for item in existing_variables:
+        for key in list(item.keys()):
+            if key not in KNOWN:
+                item.pop(key)
+    return existing_variables
```

Comment:
This function modifies the dictionaries which are part of GitLab objects. I don't think that's a good idea. How about returning a list of modified copies of `item.attributes` elements? Either by changing line 117 to
```
    existing_variables = [dict(x.attributes) for x in gitlab_variables]
```
or by doing something like `return [dict((k, v) for k, v in item.attributes.items() if k in KNOWN) for item in existing_variables]` instead.